src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { FC, useLayoutEffect, useRef, useState } from 'react';
+
+import { mergeClasses } from '../../shared/utilities';
+import styles from './slider.module.scss';
+import { SliderMarker, SliderProps } from './Slider.types';
+
+const thumbDiameter: number = +styles.thumbDiameter;
+const markerWidth: number = +styles.markerWidth;
+
+function asc(a: number, b: number) {
+    return a - b;
+}
+
+export function valueToPercent(value: number, min: number, max: number) {
+    return ((value - min) * 100) / (max - min);
+}
+
+export const Slider: FC<SliderProps> = ({
```

Comment:
Is there an advantage to forwardRef here?

Reply:
I would defer to your input here, given my limited knowledge of React. One complicating factor: because the range slider makes use of two inputs (lower value and upper value), which would take the forwardRef? If my understanding of ForwardRef is possibly off let's talk through this.

Reply:
Let's table this for now and revisit.